docs/markdown/Reference-manual.md

```diff
@@ -1509,6 +1510,20 @@ object](#build-target-object) returned by
 object](#external-program-object) returned by
 [`find_program()`](#find_program).
 
+By default, environment variable
+[`MALLOC_PERTURB_`](http://man7.org/linux/man-pages/man3/mallopt.3.html)
+is set to a random value between 1..255. This can help find memory
+leaks on configurations using glibc, including with non-GCC compilers.
+However, this can have a performance impact, and may fail a test even on
+external libraries whose internals are out of the user's control.
+If a test() unexpectedly fails, try disabling this feature by adding
+
+```meson
+nomalloc = environment({'MALLOC_PERTURB_': '0'})
+
+test(..., env: nomalloc, ...)
```

Comment:
This should make it clear that there are two different ways of disabling this _and_ that their use cases are different.

 - Disabling the perturb temporarily should be done by setting the envvar in the shell when running the test command
 - Disabling the perturb permanently should be done with the environment kwarg

It should also note that the former is the preferred approach. Tests that break when malloc perturb is used should be fixed instead if possible.